pandas/_config/localization.py

```diff
@@ -17,6 +17,29 @@
 from pandas._config.config import options
 
 
+def get_current_locale(lc_var: int = locale.LC_ALL):
+    """
+    Return the current locale associated with category ``lc_var``.
+    This is a convenience method to avoid misuses of ``getlocale``. Indeed the
+    result returned by ``getlocale`` can not always be used to set back the locale
+    using ``setlocale``. See GH#46595
+
+    Parameters
+    ----------
+    lc_var : int, default `locale.LC_ALL`
+        The category of the locale being set.
+
+    Returns
+    -------
+
+    """
+    # `getlocale` does not always play well with `setLocale`, avoid using it
+    # return locale.getlocale()
+
+    # Using `setlocale` with no second argument returns the current locale
+    return locale.setlocale(lc_var)
```

Comment:
Can we just call this directly where needed since it's a one line function? (All the notes can go at the top of the file?)

Reply:
When would we need to change this?

Reply:
if the locale python module changes behaviour. For example if the output of `setlocale(lc_var)` (get current) is less guaranteed to work with `setlocale(lc_var, l)` (set it back), than the output of `getlocale(lc_var)` (which seems more natural but unfortunately is not compliant with setting back in all cases.

Also I find it convenient as a docstring holder, to document precisely why we use and recommend this, and to refer to the associated GH issue.

But if you prefer of course I can dispatch the mod everywhere and remove this.

Reply:
Ok. In that case I recommend adding a comment on top of all of these usages explaining why we use `setlocale` and not `getlocale`.

> IMO it appears a lot of the added tests is testing Python's setlocale behavior

A single test does, actually. 

All others test that the `can_set_locale` pandas goodie does not leak in case of edge cases

Reply:
>  I recommend adding a comment on top of all of these usages explaining why we use setlocale and not getlocale.

Sure that would be good!

Reply:
Okay looks good. Thanks for sticking with it!